csrc/scheduler/matmul_utils.cpp

```diff
@@ -393,12 +394,36 @@ std::shared_ptr<MatmulParams> getMatmulHeuristics(
   // Disable magic zero for matmul kernels
   params->cparams.enable_magic_zero = false;
 
-  // Set whether to use shared memory for epilogue
   const auto& roles_map_opt =
       mma_utils::getTensorsRoles(fusion, mulSum.front().insouts);
   NVF_ERROR(roles_map_opt.isValid(), "Tensor roles map in mma is not valid.");
-
   const auto roles_map = roles_map_opt.getData();
+
+  const mma_utils::MatmulProblemLayoutOpt layout_opt =
+      mma_utils::getMmaLayout(fusion);
+  NVF_ERROR(layout_opt.isValid(), layout_opt.getErrorMsg());
```

Comment:
This fails for `CombineMulSumAsMmaTest.UseMatmulScheduler` currently. I'm looking into it.